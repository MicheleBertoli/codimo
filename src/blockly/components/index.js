/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */

function importAll(req) {
  req.keys().forEach(req);
}

// $FlowDoNotDisturb is a webpack function
importAll(require.context('./', true, /\.js$/));
