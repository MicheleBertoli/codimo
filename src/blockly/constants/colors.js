/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
export const ACTION = 345;
export const CONTAINER = 120;
