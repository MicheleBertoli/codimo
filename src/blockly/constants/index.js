/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
export * as colors from './colors';
export * as blockNames from './blockNames';
