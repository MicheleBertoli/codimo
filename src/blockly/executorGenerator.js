/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * This tester must evaluate every component.
 *
 * @flow
 */

// FIXME This function need to be more generic
// Right now only works for one specific function
function componentExecutionConfig(blockName: string) {
  // FIXME hardcoded 0
  Blockly.JavaScript[blockName] = () => (`0,${blockName};`);
}

export type ActorsToActions = Map<number, Array<string>>;
export type Executor = {|
  addBlockExecutor(blockName: string): void,
  parseInstructions(code: string): ActorsToActions,
|};
/**
 * This object is responsible for executing the code-as-string generated by blockly.
 * It allows to add every function used by each block.
 *
 * @return {Object} excecutor
 */
const executorGenerator = (): Executor => ({
  addBlockExecutor(blockName: string) {
    componentExecutionConfig(blockName);
  },
  /**
   * Each instruction follows this format:
   *      actor,action;
   * Where the semicolon is the separator.
   *
   * @param  {string} rawInstructions a sequence of raw instructions
   * @return {ActorsToActions}        a map with sequence of actions for each actor
   */
  parseInstructions: (rawInstructions: string): ActorsToActions => {
    const splittedRawInstructions = rawInstructions.split(';');
    const instructions = new Map();
    let actions;

    // Deletes the last & empty value
    splittedRawInstructions.pop();

    splittedRawInstructions.forEach((splittedRawInstruction) => {
      const elements = splittedRawInstruction.split(',');
      const key = parseInt(elements[0]);

      actions = instructions.get(key) || [];
      actions.push(elements[1]);

      instructions.set(key, actions);
    });

    return instructions;
  },
});

export default executorGenerator;
