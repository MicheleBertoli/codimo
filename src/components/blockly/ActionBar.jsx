/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

import Button from './Button';

const Bar = styled.div`
  align-items: center;
  background-color: #e3e3e3;
  border: 1px solid #ddd;
  border-bottom: 0;
  display: flex;
  height: 68px;
  width: 628px;
  & * {
    margin: 0 0 0 10px;
  }
`;

type Props = {|
  isExecuting: boolean,
  isStopped: boolean,
  handleStartGame(): void,
  handleResetGame(): void,
|};
// FIXME hardcoded title
const ActionBar = ({ isExecuting, isStopped, handleStartGame, handleResetGame }: Props) => (
  <Bar>
    {isStopped ?
      <Button
        title="▶ Dale play!"
        type="green"
        handleClick={handleStartGame}
      /> :
      <Button
        disabled={isExecuting}
        title="⏹ Reseteá!"
        type="orange"
        handleClick={handleResetGame}
      />
    }
  </Bar>
);

export default ActionBar;
