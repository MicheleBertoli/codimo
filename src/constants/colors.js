/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */

const WHITE = '#fff';

export const ACTION = '#b63668';
export const CONTAINER = '#369836';

export const BUTTON_COLORS = {
  blue: {
    bgColor: '#428bca',
    border: '#0a5a9c',
    color: WHITE,
  },
  green: {
    bgColor: '#5cb85c',
    border: '#1b7c1b',
    color: WHITE,
  },
  red: {
    bgColor: '#d9534f',
    border: '#be2f2b',
    color: WHITE,
  },
  orange: {
    bgColor: '#D3823F',
    border: '#B36320',
    color: WHITE,
  },
};
