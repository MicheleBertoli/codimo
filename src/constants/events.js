/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */

/////////////////
// Pointers
/////////////////

// To normalize touch and mouse click
export const POINTER_DOWN = 'pointerdown';
// Only click
export const CLICK = 'click';
// Only touch
export const TAP = 'tap';
