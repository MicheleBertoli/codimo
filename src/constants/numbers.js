/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */

export const ZERO = 0;
export const ONE = 1;
export const TWO = 2;
export const FOUR = 4;
export const TEN = 10;
export const HALF = TWO;

export const ACTOR_MOVEMENT_DURATION = 0.5;
export const ANCHOR_CENTER = 0.5;

export const NUMERIC_LINE_BG_COLOR = 0xffd900;
export const NUMERIC_LINE_NUMBER_BG_COLOR = 0x2a2a2a;
