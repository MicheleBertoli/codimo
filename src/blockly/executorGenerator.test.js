/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * This tester must evaluate every component.
 *
 * @flow
 */
/* eslint-disable no-magic-numbers */
import 'test/BlocklyMock';

import { blockNames } from 'blockly/constants';

import executorGenerator from './executorGenerator';

describe('blockly > executorGenerator', () => {
  it('should generate a Map of actions', () => {
    const blockData = {
      number: 0,
    };
    const executor = executorGenerator();

    executor.addBlockExecutor(blockNames.MOVE_BACKWARD);

    const code = Blockly.JavaScript[blockNames.MOVE_BACKWARD](blockData);

    expect(code).toMatchSnapshot();
    expect(Blockly.JavaScript[blockNames.MOVE_BACKWARD]).toBeInstanceOf(Function);

    const instructions = executor.parseInstructions(code + code);
    const actions = instructions.get(blockData.number);

    expect(instructions).toBeInstanceOf(Map);
    expect(instructions.size).toBe(1);
    expect(actions).toBeInstanceOf(Array);
    expect(actions).toHaveLength(2);
    // $FlowDoNotDisturb is NOT undefined
    actions.forEach(action => {
      expect(action).toBe(blockNames.MOVE_BACKWARD);
    });
  });
});
