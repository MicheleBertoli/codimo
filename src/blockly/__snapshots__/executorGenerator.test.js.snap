// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blockly > executorGenerator should generate a Map of actions 1`] = `"0,move_backward;"`;
